src/Uno.UI/UI/Xaml/Controls/Border/BorderLayerRenderer.skia.cs

```diff
@@ -0,0 +1,223 @@
+			Thickness borderThickness,
+			Brush borderBrush,
+			CornerRadius cornerRadius)
+		{
+			var disposables = new CompositeDisposable();
+			var subVisuals = new List<Visual>();
+
+			var adjustedLineWidth = borderThickness.Top;
+			var adjustedLineWidthOffset = adjustedLineWidth / 2;
+
+			var adjustedArea = area;
+			adjustedArea.Inflate(-adjustedLineWidthOffset, -adjustedLineWidthOffset);
+
+			var compositor = parent.Compositor;
+
+			var shapeVisual = compositor.CreateShapeVisual();
+			parent.Children.InsertAtBottom(shapeVisual);
+
+			var spriteShape = compositor.CreateSpriteShape();
+
+			SkiaGeometrySource2D BuildGeometry()
+			{
+				var maxRadius = Math.Max(0, Math.Min((float)area.Width / 2 - adjustedLineWidthOffset, (float)area.Height / 2 - adjustedLineWidthOffset));
+				cornerRadius = new CornerRadius(
+					Math.Min(cornerRadius.TopLeft, maxRadius),
+					Math.Min(cornerRadius.TopRight, maxRadius),
+					Math.Min(cornerRadius.BottomRight, maxRadius),
+					Math.Min(cornerRadius.BottomLeft, maxRadius));
+
+				var geometry = new SkiaGeometrySource2D();
+
+				Brush.AssignAndObserveBrush(borderBrush, color => spriteShape.StrokeBrush = compositor.CreateColorBrush(color))
+					.DisposeWith(disposables);
+
+				geometry.Geometry.MoveTo((float)adjustedArea.GetMidX(), (float)adjustedArea.Y);
+				geometry.Geometry.ArcTo((float)adjustedArea.Right, (float)adjustedArea.Top, (float)adjustedArea.Right, (float)adjustedArea.GetMidY(), (float)cornerRadius.TopRight);
+				geometry.Geometry.ArcTo((float)adjustedArea.Right, (float)adjustedArea.Bottom, (float)adjustedArea.GetMidX(), (float)adjustedArea.Bottom, (float)cornerRadius.BottomRight);
+				geometry.Geometry.ArcTo((float)adjustedArea.Left, (float)adjustedArea.Bottom, (float)adjustedArea.Left, (float)adjustedArea.GetMidY(), (float)cornerRadius.BottomLeft);
+				geometry.Geometry.ArcTo((float)adjustedArea.Left, (float)adjustedArea.Top, (float)adjustedArea.GetMidX(), (float)adjustedArea.Top, (float)cornerRadius.TopLeft);
+
+				geometry.Geometry.Close();
+
+				if (background is LinearGradientBrush lgbBackground)
+				{
+					//var fillMask = new CAShapeLayer()
+					//{
+					//	Path = path,
+					//	Frame = area,
+					//	// We only use the fill color to create the mask area
+					//	FillColor = _Color.White.CGColor,
+					//};
+
+					//// We reduce the adjustedArea again so that the gradient is inside the border (like in Windows)
+					//adjustedArea = adjustedArea.Shrink((nfloat)adjustedLineWidthOffset);
+
+					//CreateLinearGradientBrushLayers(area, adjustedArea, parent, sublayers, ref insertionIndex, lgbBackground, fillMask);
+				}
+				else if (background is SolidColorBrush scbBackground)
+				{
+					Brush.AssignAndObserveBrush(scbBackground, color => spriteShape.FillBrush = compositor.CreateColorBrush(color))
+						.DisposeWith(disposables);
+				}
+				else if (background is ImageBrush imgBackground)
+				{
+					//var uiImage = imgBackground.ImageSource?.ImageData;
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5872302)